django/db/backends/base/schema.py

```diff
@@ -1369,14 +1369,9 @@ def _field_indexes_sql(self, model, field):
             output.append(self._create_index_sql(model, fields=[field]))
         return output
 
-    def _field_should_be_altered(self, old_field, new_field):
-        _, old_path, old_args, old_kwargs = old_field.deconstruct()
-        _, new_path, new_args, new_kwargs = new_field.deconstruct()
-        # Don't alter when:
-        # - changing only a field name
-        # - changing an attribute that doesn't affect the schema
-        # - adding only a db_column and the column name is not changed
-        non_database_attrs = [
+    @property
+    def _field_non_database_attrs(self):
```

Comment:
What do you think about moving this to the `Field()` per [Adam's comment](https://code.djangoproject.com/ticket/33471#comment:8)? e.g.
```python
class Field(RegisterLookupMixin):

    ...
    @cached_property
    def non_db_attrs(self):
         """
         Return the list attributes that don't affect a column definition.
         These attributes are ignored when altering the field.
         """
         return [
            "blank",
            "choices",
            "db_column",
            "editable",
            "error_messages",
            "help_text",
            "limit_choices_to",
            # Database-level options are not supported, see #21961.
            "on_delete",
            "related_name",
            "related_query_name",
            "validators",
            "verbose_name",
        ]
```

\cc @adamchainz

Reply:
I agree where I defined it is perhaps not where it belongs. I have moved it to be part of the database features as an alternative, maybe better than before?

In order to get the field implementation to work, @felixxm sorry to ask, could you help direct me as to:
- how I would overwrite the `non_db_attrs` in the sqllite3 case
- within the `_field_should_be_altered` function, I need to reference the `non_db_attrs`. There are both an `old_field` and a `new_field` in this function, would I pick one of these in order to be the `non_db_attrs` or would it be better to loop through them separately (unsure if that makes sense)

Reply:
> `how I would overwrite the non_db_attrs in the sqllite3 case`

This doesn't need to be overwrite for SQLite when we move it to the field, because custom fields as `EnumField` from `django-mysql` will be able to remove `"choices"` on their own from `EnumField.non_db_attrs`, so we could add `"choices"` to the `Field.non_db_attrs` for all databases.

> within the _field_should_be_altered function, I need to reference the non_db_attrs. There are both an old_field and a new_field in this function, would I pick one of these in order to be the non_db_attrs or would it be better to loop through them separately (unsure if that makes sense)

We should do this separately because a field type can change.

Reply:
I have updated :tada: thank you for explaining. :star: